src/App.tsx

```diff
@@ -15,35 +15,96 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortByName {
+  Default = '',
+  Abc = 'abc',
+  Length = 'length',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, reverse }: { sortField: SortByName; reverse: boolean },
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      if (sortField === SortByName.Abc) {
+        return good1.localeCompare(good2);
+      }
+
+      if (sortField === SortByName.Length) {
+        return good1.length - good2.length;
+      }
+
+      return 0;
+    });
+  }
+
+  if (reverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortName, setSortName] = useState<SortByName>(SortByName.Default);
+  const [reverse, setReverse] = useState(false);
+
+  const newGoods = getPreparedGoods(goodsFromServer, {
+    sortField: sortName,
+    reverse,
+  });
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          onClick={() => setSortName(SortByName.Abc)}
+          type="button"
+          className={`button is-info ${sortName !== SortByName.Abc && 'is-light'}`}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          onClick={() => setSortName(SortByName.Length)}
+          type="button"
+          className={`button is-success ${sortName !== SortByName.Length && 'is-light'}`}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          onClick={() => setReverse(!reverse)}
+          type="button"
+          className={`button is-warning ${!reverse && 'is-light'}`}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortName || reverse) && (
+          <button
+            onClick={() => {
+              setSortName(SortByName.Default);
+              setReverse(false);
+            }}
```

Comment:
Move this logic from jsx to the helper reset function and use it here
```suggestion
            onClick={reset}
```